modules/mono/editor/GodotTools/GodotTools/CsTranslationParserPlugin.cs

```diff
@@ -0,0 +1,430 @@
+                    commentText = commentText.TrimPrefix(TranslationCommentPrefix).Trim();
+                }
+                else if (commentText == NoTranslateComment || commentText.StartsWith(NoTranslateComment + ":"))
+                {
+                    skip = true;
+                }
+            }
+            else
+            {
+                // Parse multiline comment
+                for (var index = line - 1; index >= 0; index--)
+                {
+                    var multilineCommentData =
+                        comments.FirstOrDefault(comment => comment.EndLine == index && comment.Newline);
+                    if (multilineCommentData == null)
+                    {
+                        commentText = "";
+                        break;
+                    }
+                    // multiline comment
+                    if (multilineCommentData.StartLine != multilineCommentData.EndLine)
+                    {
+                        var multilineComments = multilineCommentData.Comment.TrimSuffix("*/").Trim().Split("\n")
+                            .Select(lineStr => lineStr.TrimPrefix("/*").Trim().TrimPrefix(TranslationCommentPrefix));
+                        commentText = string.Join("\n", multilineComments);
+                        if (commentText == NoTranslateComment || commentText.StartsWith(NoTranslateComment + ":"))
+                        {
+                            commentText = "";
+                            skip = true;
+                        }
+                        break;
+                    }
+                    // multiline single line comment
+                    var currentComment = multilineCommentData.Comment.TrimStart('/').Trim();
+                    if (currentComment == "") continue;
+                    if (commentText == "")
+                    {
+                        commentText = currentComment;
+                    }
+                    else
+                    {
+                        commentText = currentComment + "\n" + commentText;
+                    }
+                    if (currentComment.StartsWith(TranslationCommentPrefix))
+                    {
+                        commentText = commentText.TrimPrefix(TranslationCommentPrefix).Trim();
+                        break;
+                    }
+                    if (currentComment == NoTranslateComment || currentComment.StartsWith(NoTranslateComment + ":"))
+                    {
+                        commentText = "";
+                        skip = true;
+                        break;
+                    }
+                }
+            }
+
+            SymbolInfo? symbolInfo = null;
+            if (invocation.Expression is IdentifierNameSyntax identifierNameSyntax)
+            {
+                symbolInfo = semanticModel.GetSymbolInfo(identifierNameSyntax);
+            }
+            if (invocation.Expression is MemberAccessExpressionSyntax memberAccessExpressionSyntax &&
```

Comment:
Not a required change, but this can be less verbose written as follows.
```cs
if (invocation.Expression is MemberAccessExpressionSyntax { Name: IdentifierNameSyntax nameSyntax })
```
There are a couple of places where pattern matching can be simplified in similar ways 🤓